scipy/stats/_qmc.py

```diff
@@ -481,6 +485,186 @@ def _perturb_discrepancy(sample: np.ndarray, i1: int, i2: int, k: int,
     return disc_ep
 
 
+def _jitter_sample(
+        sample: np.ndarray,
+        rng: Union[np.random.Generator, np.random.RandomState],
+        eps: float = 1e-10
+) -> np.ndarray:
+    """Randomly jitter `sample` until they are unique.
+
+    If the samples are not unique, the number of regions in our
+    tessellation will be less than the number of input samples.
+    """
+    vals, count = np.unique(sample, return_counts=True)
+    if np.any(vals[count > 1]):
+        offset = rng.uniform(-eps, eps, size=(len(sample), sample.shape[1]))
+        sample = sample * (1 + offset)
+    return sample
```

Comment:
At the moment, none of the tests added here actually require this--for example the test suite passes if I simply remove the usage of the jittering function:

```diff
diff --git a/scipy/stats/_qmc.py b/scipy/stats/_qmc.py
index 2552bc5ab..57af783c4 100644
--- a/scipy/stats/_qmc.py
+++ b/scipy/stats/_qmc.py
@@ -535,7 +535,6 @@ def _lloyd_centroidal_voronoi_tessellation(
         The sample after an iteration of Lloyd's algorithm.
 
     """
-    sample = _jitter_sample(sample, rng=rng)
     new_sample = np.empty_like(sample)
 
     voronoi = spatial.Voronoi(sample)
```

What I'm really wondering though, is if we really need our own jittering function when Qhull already has the QJ option to "joggle input" when points are too close together.

Reply:
Oh great, then yes I guess it would be better to use these options. I am not sure how it works. Shall I just add QJ? Or shall I add the default describe in the doc on top?

Reply:
Not sure--in `spatial` we tend to specify a few default Qhull options but also give the user the option to add their own. I guess you already had joggling turned on by default here. I know we've had a few complaints from users who basically wanted to be able to have full control over the Qhull options, but not too many.

Reply:
I assume that this comment is addressed by `qhull_options`? 

If so, `qhull_options` needs to be explained more completely. The only documentation that refers to QHull is

```
    qhull_options : str, optional
        Additional options to pass to Qhull. See Qhull manual
        for details. (Default: "Qbb Qc Qz Qj Qx" for ndim > 4 and
        "Qbb Qc Qz Qj" otherwise.)
```

I think we need to state which QHull routine is used (and for what).